ledger/src/blockstore_processor.rs

```diff
@@ -326,11 +341,22 @@ fn execute_batches(
 ) -> Result<()> {
     let lock_results = batches
         .iter()
-        .flat_map(|batch| batch.lock_results().clone())
+        .flat_map(|TransactionBatchWithIndexes { batch, .. }| batch.lock_results().clone())
         .collect::<Vec<_>>();
     let sanitized_txs = batches
         .iter()
-        .flat_map(|batch| batch.sanitized_transactions().to_vec())
+        .flat_map(|TransactionBatchWithIndexes { batch, .. }| {
+            batch.sanitized_transactions().to_vec()
+        })
+        .collect::<Vec<_>>();
+    let transaction_indexes = batches
+        .iter()
+        .flat_map(
+            |TransactionBatchWithIndexes {
+                 transaction_indexes,
+                 ..
+             }| transaction_indexes.to_vec(),
+        )
```

Comment:
hmmm I feel like we could allocate the vector ahead of time since we know the size (the total num transactions we already did a sum earlier) and then merge all this stuff into one loop rather than iterating multiple times

Reply:
Maybe we could even just push all of this flattened stuff earlier in the previous function here: https://github.com/solana-labs/solana/blob/68df61faf6defea65a58efac5a0aca157a5b3774/ledger/src/blockstore_processor.rs#L563-L566 and just pass the flattened items here in a struct `ReplayableTransactions`